cloud/scope/machinepoolmachine.go

```diff
@@ -318,18 +320,46 @@ func (m *MachinePoolMachineScope) HasLatestModelApplied(_ context.Context, insta
 	// Get the image from the disk URL path to compare with the latest image name
 	diskImage, err := url.Parse(instance.SourceImage)
 	if err != nil {
+		log.Error(err, "Error Parsing the Instance.SourceImage from the disk attached.")
 		return false, err
 	}
 	instanceImage := path.Base(diskImage.Path)
 
-	// Check if the image is the latest
-	if image == instanceImage {
+	// Check if the image is the latest and additionalLabels
+	hasAdditionalLabelsDiff, err := m.hasAdditionalLabelsDiff(ctx, labels)
+	if err != nil {
+		log.Error(err, "Error checking the AdditonalLabels")
+		return false, err
+	}
+	if image == instanceImage && hasAdditionalLabelsDiff {
```

Comment:
I think the name `hasAdditionalLabelsDiff` is a bit confusing here. Ideally it should mean `doesNotHaveAdditionalLabelsDiff` in the above context, correct?

Reply:
just added a change here https://github.com/newrelic-forks/cluster-api-provider-gcp/pull/25/commits/761d5d79ea1bc944daf93a0a4fbab5af0249208b, please check